src/Type.cc

```diff
@@ -603,6 +614,9 @@ SetType::~SetType() = default;
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 FuncType::Capture::Capture(detail::IDPtr _id, bool _deep_copy) : id(std::move(_id)), deep_copy(_deep_copy) {
     is_managed = id ? ZVal::IsManagedType(id->GetType()) : false;
+    if ( ! is_managed )
+        // For non-managed types, deep copying isn't applicable.
+        deep_copy = false;
```

Comment:
Is this a bug fix that we should pick-up for Zeek 7.0 or early?

Reply:
It's not needed for what shipped in Zeek 7.0, but it matters for some upcoming refinements in how lambdas are dealt with for script optimization